### PR TITLE
Fixes #1780 Clone of chart does not clone deviation lines

### LINKED ISSUE #1780: Clone of chart does not clone deviation lines
@georgeDaskalakis @rwmcintosh 
There is a method in chart called UpdatePropertiesFrom (sourcexx)

This method need to be implemented in all objects having state .This is used in cloning etc. 
I think this is missing for the Chart with folds being saved. I have fixed the issue with closing in PKSim but this is missing in core
Can you tale a look please?

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

File: src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+      public void AddToDeviationFoldValue(float foldValue)
+      {
+         if (DeviationFoldValues.Contains(foldValue))
+            return;
+
+         DeviationFoldValues.Add(foldValue);
+      }
+
+      public override void UpdatePropertiesFrom(IUpdatable source, ICloneManager cloneManager)
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs:42]: The missing method transfers the foldValues to the new chart but that introduced an issue where if the foldValue was present in the list then the service assumed the curve was present in the chart, but that's not necessarily the case now

File: src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      {
+         base.UpdatePropertiesFrom(source, cloneManager);
+         var sourceChart = source as PredictedVsObservedChart;
+
+         sourceChart?.DeviationFoldValues.Each(AddToDeviationFoldValue);
+      }
+
+      public bool HasDeviationCurveFor(float foldValue)
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs:50]: The cache keeps track of foldValues for which there is a curve. If the foldValue is in the list and no value in the cache then it needs to be plotted

File: src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            return;
+
+         DeviationFoldValues.Add(foldValue);
+      }
+
+      public override void UpdatePropertiesFrom(IUpdatable source, ICloneManager cloneManager)
+      {
+         base.UpdatePropertiesFrom(source, cloneManager);
+         var sourceChart = source as PredictedVsObservedChart;
+
```
Comment (msevestre) [inline src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs:46]: typically we exit if the object is not of the right type

File: src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Core.Extensions;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Core.Chart
+{
+   public abstract class PredictedVsObservedChart : AnalysisChart
+   {
+      private readonly Cache<float, IEnumerable<DataRepository>> _deviationRepositoryCache = new Cache<float, IEnumerable<DataRepository>>(onMissingKey: x => null);
```
Comment (msevestre) [inline src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs:16]: is there a reason why you are not storing an IReadOnlyList here?

**Review by msevestre — COMMENTED** [review]
looks good to me. Just one remark for code sync with other UpdatePropertiesFrom and a question regarding usage of IEnumerable

File: src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs (hunk 4 above)
Comment (rwmcintosh) [reply src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs:16]: I need to know for each foldValue, if there was a data repository created. That's how I can tell the difference between what the chart is supposed to have, vs what it already has.

Comment (rwmcintosh) [reply src/OSPSuite.Core/Chart/PredictedVsObservedChart.cs:16]: oh, I see what you mean. forget my comment. above

**Pull request merged**